packages/db/src/lib/granule.ts

```diff
@@ -45,3 +48,39 @@ export const upsertGranuleWithExecutionJoinRecord = async (
   }
   return [granuleCumulusId];
 };
+
+/**
+ * Get cumulus IDs for all executions associated to a set of granules
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction -
+ *  DB client or transaction
+ * @param {Array<Object>} granules - array of granules with collectionId and granuleId
+ * @param {Object} [collectionPgModel] - Collection PG model class instance
+ * @param {Object} [granulePgModel] - Granule PG model class instance
+ * @param {Object} [granulesExecutionsPgModel]
+ *   Granules/executions PG model class instance
+ * @returns {Promise<number[]>}
+ */
+export const getGranuleExecutionCumulusIds = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  granules: Array<{ collectionId: string, granuleId: string }>,
+  collectionPgModel = new CollectionPgModel(),
+  granulePgModel = new GranulePgModel(),
+  granulesExecutionsPgModel = new GranulesExecutionsPgModel()
+): Promise<Array<number>> => {
+  const granuleCumulusIds: Array<number> = await Promise.all(granules.map(async (granule) => {
+    const { name, version } = deconstructCollectionId(granule.collectionId);
+    return await granulePgModel.getRecordCumulusId(knexOrTransaction, {
+      granule_id: granule.granuleId,
+      collection_cumulus_id: await collectionPgModel.getRecordCumulusId(
```

Comment:
Sort of a nit, but it **could** be expensive to run this query for every collection depending on the number of granules. Can we create a hash map to store the `cumulus_id` of collections for each name/version combo and only look it up if we didn't already?

Reply:
Yeah that's a really good call